supervisor/os/disk.py

```diff
@@ -0,0 +1,47 @@
+"""Home Assistant Operating-System DataDisk."""
+import logging
+from pathlib import Path
+from typing import Optional
+
+from ..coresys import CoreSys, CoreSysAttributes
+from ..exceptions import DBusError, HassOSError, HassOSJobError, HostError
+from ..jobs.const import JobCondition, JobExecutionLimit
+from ..jobs.decorator import Job
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class DataDisk(CoreSysAttributes):
+    """Handle DataDisk feature from OS."""
+
+    def __init__(self, coresys: CoreSys) -> None:
+        """Initialize DataDisk."""
+        self.coresys = coresys
+
+    @property
+    def disk_used(self) -> Optional[Path]:
+        """Return Path to used Disk for data."""
+        return self.sys_dbus.agent.datadisk.current_device
+
+    @Job(
+        conditions=[JobCondition.HAOS, JobCondition.HOST_AGENT, JobCondition.HEALTHY],
+        limit=JobExecutionLimit.ONCE,
+        on_condition=HassOSJobError,
+    )
+    async def migrate_disk(self, new_disk: Path) -> None:
+        """Move data partition to a new disk."""
+        # Need some error handling, but we need know what disk_used will return
+        try:
+            await self.sys_dbus.agent.datadisk.change_device(new_disk)
+        except DBusError as err:
+            raise HassOSError(
+                f"Can't move data partition to {new_disk!s}: {err!s}", _LOGGER.error
+            ) from err
+
+        try:
+            await self.sys_host.control.reboot()
```

Comment:
Can the system be used at all after `self.sys_dbus.agent.datadisk.change_device` is called?
Or are a reboot always needed directly after?

Reply:
> are a reboot always needed directly after?
yes